src/consumer.ts

```diff
@@ -287,97 +336,51 @@ export class Consumer extends EventEmitter {
-      });
+
+    if (this.batchSize > 0) {
+      const receiveParams = {
+        QueueUrl: this.queueUrl,
+        AttributeNames: this.attributeNames,
+        MessageAttributeNames: this.messageAttributeNames,
+        MaxNumberOfMessages: this.batchSize,
+        WaitTimeSeconds: this.waitTimeSeconds,
+        VisibilityTimeout: this.visibilityTimeout
+      };
+
+      this.receiveMessage(receiveParams)
+        .then(this.handleSqsResponse)
+        .catch(err => {
+          this.emit('error', err);
+          if (isConnectionError(err)) {
+            debug('There was an authentication error. Pausing before retrying.');
+            currentPollingTimeout = this.authenticationErrorTimeout;
+          }
+          return;
+        })
+        .then(() => {
+          setTimeout(this.poll, currentPollingTimeout);
+        })
+        .catch(err => {
+          this.emit('error', err);
+        });
+    } else {
+      setTimeout(this.poll, currentPollingTimeout);
+    }
   }
 
   private async processMessageBatch(messages: SQSMessage[]): Promise<void> {
     messages.forEach(message => {
       this.emit('message_received', message);
     });
 
-    try {
-      await this.executeBatchHandler(messages);
-      await this.deleteMessageBatch(messages);
-      messages.forEach(message => {
-        this.emit('message_processed', message);
-      });
-    } catch (err) {
-      this.emit('error', err, messages);
-
-      if (this.terminateVisibilityTimeout) {
-        try {
-          await this.terminateVisabilityTimeoutBatch(messages);
-        } catch (err) {
-          this.emit('error', err, messages);
-        }
-      }
-    }
-  }
-
-  private async deleteMessageBatch(messages: SQSMessage[]): Promise<void> {
-    debug('Deleting messages %s', messages.map(msg => msg.MessageId).join(' ,'));
-
-    const deleteParams = {
-      QueueUrl: this.queueUrl,
-      Entries: messages.map(message => ({
-        Id: message.MessageId,
-        ReceiptHandle: message.ReceiptHandle
-      }))
-    };
-
-    try {
-      await this.sqs.deleteMessageBatch(deleteParams).promise();
-    } catch (err) {
-      throw toSQSError(err, `SQS delete message failed: ${err.message}`);
-    }
-  }
-
-  private async executeBatchHandler(messages: SQSMessage[]): Promise<void> {
-    try {
-      await this.handleMessageBatch(messages);
-    } catch (err) {
-      err.message = `Unexpected message handler failure: ${err.message}`;
-      throw err;
-    }
-  }
-
-  private async terminateVisabilityTimeoutBatch(messages: SQSMessage[]): Promise<PromiseResult<any, AWSError>> {
-    const params = {
-      QueueUrl: this.queueUrl,
-      Entries: messages.map(message => ({
-        Id: message.MessageId,
-        ReceiptHandle: message.ReceiptHandle,
-        VisibilityTimeout: 0
-      }))
-    };
-    return this.sqs.changeMessageVisibilityBatch(params).promise();
+    this.reportNumberOfMessagesReceived(messages.length);
+    this.handleMessageBatch(messages, this);
```

Comment:
why don't we use await on this method? as it is async

Reply:
I want the call to be sync, but it is a quesiton for other developers, i want to use it in automations as something that does
eachMessage.map(message=> message.doSomehting.then.catch)